### PR TITLE
update EditMenuProductFragment.kt

### PR DESCRIPTION
**Pull request by xidealo**

### CODE REVIEW

File: domain/src/main/java/com/bunbeauty/domain/feature/order_list/GetMenuSortedByVisibleDescUseCase.kt
Comment (MaxAstin) [inline domain/src/main/java/com/bunbeauty/domain/feature/order_list/GetMenuSortedByVisibleDescUseCase.kt:1]: Remove this file

File: domain/src/main/java/com/bunbeauty/domain/use_case/GetMenuProductUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import com.bunbeauty.domain.model.menu_product.MenuProduct
+import com.bunbeauty.domain.repo.MenuProductRepo
+import javax.inject.Inject
+
+class GetMenuProductUseCase @Inject constructor(
+    private val menuProductRepo: MenuProductRepo
+) {
+    suspend operator fun invoke(menuProductUuid: String): MenuProduct {
+        return menuProductRepo.getMenuProduct(menuProductUuid = menuProductUuid)
+            ?: throw Exception("")
```
Comment (MaxAstin) [inline domain/src/main/java/com/bunbeauty/domain/use_case/GetMenuProductUseCase.kt:12]: Add more suitable Exception or message

File: domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductDescriptionException.kt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package com.bunbeauty.domain.exception.updateproduct
+
+object MenuProductDescriptionException : Exception()
```
Comment (MaxAstin) [inline domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductDescriptionException.kt:3]: replace `object` with a `class`

File: domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductNameException.kt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package com.bunbeauty.domain.exception.updateproduct
+
+object MenuProductNameException : Exception()
```
Comment (MaxAstin) [inline domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductNameException.kt:3]: replace `object` with a `class`

File: domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductNewPriceException.kt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package com.bunbeauty.domain.exception.updateproduct
+
+object MenuProductNewPriceException : Exception()
```
Comment (MaxAstin) [inline domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductNewPriceException.kt:3]: replace `object` with a `class`

File: domain/src/test/java/com/bunbeauty/domain/GetMenuUseCaseTest.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import javax.inject.Inject
+import kotlin.test.BeforeTest
+import kotlin.test.assertTrue
+import kotlin.test.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class GetMenuUseCaseTest {
```
Comment (MaxAstin) [inline domain/src/test/java/com/bunbeauty/domain/GetMenuUseCaseTest.kt:14]: Update this test or remove it

File: domain/src/test/java/com/bunbeauty/domain/UpdateMenuProductUseCaseTest.kt [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import com.bunbeauty.domain.repo.MenuProductRepo
+import com.bunbeauty.domain.use_case.UpdateMenuProductUseCase
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class UpdateMenuProductUseCaseTest {
```
Comment (MaxAstin) [inline domain/src/test/java/com/bunbeauty/domain/UpdateMenuProductUseCaseTest.kt:13]: Update this test or remove it

File: presentation/build.gradle.kts [hunk 7]
```diff
@@ -45,4 +45,18 @@ dependencies {
     // Hilt
     implementation(Dagger.hilt)
     kapt(Dagger.hiltCompiler)
+    implementation("javax.inject:javax.inject:1")
+
+    //DataStore
+    implementation(DataStore.dataStorePreferences)
```
Comment (MaxAstin) [inline presentation/build.gradle.kts:51]: DataStore to presentation layer?

File: presentation/src/main/java/com/bunbeauty/presentation/view_model/menu/edit_menu_product/EditMenuProductViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+                            newPrice = mutableState.value.newPrice.toIntOrNull() ?: 0,
+                            oldPrice = mutableState.value.oldPrice
+                                .ifEmpty { null }
+                                ?.toInt() ?: 0,
+                            utils = if (mutableState.value.nutrition.isEmpty()) {
+                                mutableState.value.utils
+                            } else {
+                                ""
+                            },
+                            nutrition = mutableState.value.nutrition.toIntOrNull(),
```
Comment (MaxAstin) [inline presentation/src/main/java/com/bunbeauty/presentation/view_model/menu/edit_menu_product/EditMenuProductViewModel.kt:88]: Null nutrition will not update value, lets also add `?: 0` and make a rework on backend

File: presentation/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -76,10 +76,29 @@
 
     <!-- EDIT MENU PRODUCT -->
     <string name="error_edit_menu_product_empty_name">Название не может быть пустым</string>
+    <string name="error_edit_menu_product_empty_description">Описание не может быть пустым</string>
```
Comment (MaxAstin) [inline presentation/src/main/res/values/strings.xml:79]: Move everything possible to the app module

**Review by MaxAstin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menu/EditMenuProductFragment.kt [hunk 10]
```diff
@@ -2,108 +2,320 @@ package com.bunbeauty.fooddeliveryadmin.screen.menu
...
+                modifier = Modifier.padding(top = 8.dp),
+                checked = state.isVisible,
+                onCheckChanged = viewModel::onVisibleChanged,
+                labelStringId = R.string.title_edit_menu_product_is_visible,
+                enabled = !state.isLoadingButton
+            )
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            LoadingButton(
```
[the remark this answers is not in the document]
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menu/EditMenuProductFragment.kt:257]: ?

[the remark this answers is not in the document]
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menu/EditMenuProductFragment.kt:257]: need help with flag

File: domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductDescriptionException.kt (hunk 2 above)
Comment (xidealo) [reply domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductDescriptionException.kt:3]: ?

File: domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductNameException.kt (hunk 3 above)
Comment (xidealo) [reply domain/src/main/java/com/bunbeauty/domain/exception/updateproduct/MenuProductNameException.kt:3]: ?

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menu/MenuViewModel.kt [hunk 11]
```diff
@@ -2,52 +2,60 @@ package com.bunbeauty.fooddeliveryadmin.screen.menu
...
 import kotlinx.coroutines.CoroutineExceptionHandler
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class MenuViewModel @Inject constructor(
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menu/MenuViewModel.kt:22]: I guess it's time to move this VM to ` :presentation`

File: presentation/src/main/java/com/bunbeauty/presentation/feature/order/mapper/OrderProductMapper.kt [hunk 12]
```diff
@@ -12,7 +12,7 @@ class OrderProductMapper @Inject constructor(
 
     fun map(orderProduct: OrderProduct): OrderDetailsUiState.Product {
         return OrderDetailsUiState.Product(
-            title = orderProduct.comboDescription ?: orderProduct.name,
+            title = orderProduct.name + "\n" + (orderProduct.comboDescription ?: ""),
```
Comment (MaxAstin) [inline presentation/src/main/java/com/bunbeauty/presentation/feature/order/mapper/OrderProductMapper.kt:15]: Let's add a second line only if `comboDescription` is not null or empty

**Review by MaxAstin — CHANGES_REQUESTED** [review]

**Pull request merged**